File: src/main/java/net/frozenorb/potpvp/util/menu/pagination/ViewAllPagesMenu.java
package net.frozenorb.potpvp.util.menu.pagination;

import net.frozenorb.potpvp.util.menu.buttons.BackButton;
import net.frozenorb.potpvp.util.menu.Button;
import net.frozenorb.potpvp.util.menu.Menu;
import lombok.Getter;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.Map;

@RequiredArgsConstructor
public class ViewAllPagesMenu extends Menu {

    @NonNull @Getter PaginatedMenu menu;


    @Override
    public String getTitle(Player player) {
        return "Jump to page";
    }

    @Override
    public Map<Integer, Button> getButtons(Player player) {

        HashMap<Integer, Button> buttons = new HashMap<>();

        buttons.put(0, new BackButton(menu));

        int index = 10;


        for (int i = 1; i <= menu.getPages(player); i++) {
            buttons.put(index++, new JumpToPageButton(i,this.menu));

            if ((index - 8) % 9 == 0) {
                index += 2;
            }

        }


        return buttons;
    }

    @Override
    public boolean isAutoUpdate() {
        return true;
    }

}

File: src/main/kotlin/net/frozenorb/potpvp/kt/menu/pagination/ViewAllPagesMenu.kt
package net.frozenorb.potpvp.kt.menu.pagination

import net.frozenorb.potpvp.kt.menu.Button
import net.frozenorb.potpvp.kt.menu.Menu
import net.frozenorb.potpvp.kt.menu.buttons.BackButton
import java.util.HashMap
import org.bukkit.entity.Player

class ViewAllPagesMenu(private val menu: PaginatedMenu) : Menu() {

    init {
        autoUpdate = true
    }

    override fun getTitle(player: Player): String {
        return "Jump to page"
    }

    override fun getButtons(player: Player): Map<Int, Button> {
        val buttons = HashMap<Int, Button>()

        buttons[0] = BackButton {
            menu.openMenu(player)
        }

        var index = 10
        for (i in 1..menu.getPages(player)) {
            buttons[index++] = JumpToPageButton(i, menu)
            if ((index - 8) % 9 == 0) {
                index += 2
            }
        }

        return buttons
    }

}